src/server/scripts/World/item_scripts.cpp

```diff
@@ -412,6 +412,70 @@ class item_captured_frog : public ItemScript
     }
 };
 
+enum ItemPumpkinTreat // Item 20557, Hallow's End Pumpkin Treat
+{
+    SPELL_HALLOWS_END_CANDY_ORANGE_GIANT         = 24924, // Effect 1: Apply Aura: Mod Size, Value: 30%
+    SPELL_HALLOWS_END_CANDY_SKELETON             = 24925, // Effect 1: Apply Aura: Change Model (Skeleton), 2) Apply Aura: Underwater Breathing
+    SPELL_HALLOWS_END_CANDY_PIRATE               = 24926, // Effect 1: Apply Aura: Increase Swim Speed, Value: 50%
+    SPELL_HALLOWS_END_CANDY_GHOST                = 24927, // Effect 1: Apply Aura: Levitate / Hover, 2: Apply Aura: Slow Fall, 3: Apply Aura: Water Walking
+    SPELL_HALLOWS_END_CANDY_FEMALE_DEFIAS_PIRATE = 44742, // Effect 1: Apply Aura: Change Model (Defias Pirate, Female), 2: Increase Swim Speed, Value: 50%
+    SPELL_HALLOWS_END_CANDY_MALE_DEFIAS_PIRATE   = 44743  // Effect 1: Apply Aura: Change Model (Defias Pirate), 2: Increase Swim Speed, Value: 50%
+};
+
+class item_pumpkin_treat : public ItemScript
+{
+    public:
+
+        item_pumpkin_treat() : ItemScript("item_pumpkin_treat") { }
+
+        bool OnUse(Player* player, Item* /*item*/, SpellCastTargets const& /*targets*/)
+        {
+            if (player->HasAura(SPELL_HALLOWS_END_CANDY_ORANGE_GIANT))
+                player->RemoveAura(SPELL_HALLOWS_END_CANDY_ORANGE_GIANT);
```

Comment:
SPELL_HALLOWS_END_CANDY_ORANGE_GIANT is not "Aura" object, is a enum right?
Why not use ```player-RemoveAurasDueToSpell(SPELL_HALLOWS_END_CANDY_ORANGE_GIANT);```